From 657811fa84c920e269ac53c024df3f406a29d3f6 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:02 +0800
Subject: [PATCH 026/150] dt-bindings: Add bindings for Phytium I2C/SMBus
 controller

This patch documents the DT binding for the Phytium I2C controller
with SMBus alert interrupt support.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: Id6d6a1f0f727ebb02df565247b76ce1ccd5b1b10
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../devicetree/bindings/i2c/phytium,i2c.yaml  | 51 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 52 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/i2c/phytium,i2c.yaml

diff --git a/Documentation/devicetree/bindings/i2c/phytium,i2c.yaml b/Documentation/devicetree/bindings/i2c/phytium,i2c.yaml
new file mode 100644
index 000000000000..992a4d7cbb47
--- /dev/null
+++ b/Documentation/devicetree/bindings/i2c/phytium,i2c.yaml
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/i2c/snps,designware-i2c.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium I2C/SMBus Controller
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+allOf:
+  - $ref: /schemas/i2c/i2c-controller.yaml#
+
+properties:
+  compatible:
+    const: phytium,i2c
+
+  reg:
+    minItems: 1
+    items:
+      - description: Offset and length of the memory mapped registers
+
+  interrupts:
+    maxItems: 1
+
+  interrupt-names:
+    const: smbus_alert
+    description: should be "smbus_alert" if SMBus alert interrupt is supported
+
+  clocks:
+    minItems: 1
+    items:
+      - description: I2C controller reference clock source
+
+unevaluatedProperties: false
+
+required:
+  - compatible
+  - reg
+  - interrupts
+
+examples:
+  - |
+    i2c0: i2c@28011000 {
+      compatible = "phytium,i2c";
+      reg = <0x0 0x28011000 0x0 0x1000>;
+      interrupts = <GIC_SPI 89 IRQ_TYPE_LEVEL_HIGH>;
+      interrupt-names = "smbus_alert";
+      clocks = <&sysclk_48mhz>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 7b1b8529ccf3..a87ad9aa63e2 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2531,6 +2531,7 @@ M:	Chen Baozi <chenbaozi@phytium.com.cn>
 S:	Maintained
 W:	https://www.phytium.com.cn
 F:	Documentation/devicetree/bindings/gpio/phytium,gpio.yaml
+F:	Documentation/devicetree/bindings/i2c/phytium,i2c.yaml
 F:	Documentation/devicetree/bindings/mmc/phytium,sdci.yaml
 F:	Documentation/devicetree/bindings/net/can/phytium,can.yaml
 F:	Documentation/devicetree/bindings/sound/phytium,hda.yaml
-- 
2.47.0

